Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy.swift

```diff
@@ -508,6 +508,11 @@ struct PathHierarchy {
             // If a parent ID was provided, start at that node and continue up the hierarchy until that node has a child that matches the first path components name.
             var parentNode = lookup[parentID]!
             let firstComponent = remaining.first!
+            // Check if the start node has a child that matches the first path components name.
+            if parentNode.children.keys.contains(firstComponent.name) || parentNode.children.keys.contains(firstComponent.full) {
```

Comment:
Overall it looks good, seems like the right thing to do. Beyond the change made in this radar, i see parentNode has ! which can lead to failure, is it worth adding a guard.

Reply:
Node identifiers can only be created by the PathHierarchy and it adds every node into the lookup. So since the identifier exist it's safe to unwrap the node.